### ISSUE TITLE
[Lab1, GaussianNB/Βήμα 15] Μηδενικό variance

### ISSUE THREAD

**Issue by foivospar** [slp-ntua/patrec-labs#13]
Καλησπέρα,

Υπάρχουν pixel (π.χ. στο λευκό background των ψηφίων) με ίδια grayscale τιμή για όλα τα training samples, δηλαδή μηδενικό variance. Πώς να χειριστούμε αυτές τις περιπτώσεις; Να θεωρήσουμε κρουστική probability density (και πως θα προσεγγίσουμε το άπειρο στο x=mean για συνεχή feature); Στο documentation της GaussianNB του sklearn, βλέπω πως υπάρχει η παράμετρος var_smoothing (default=1e-9) που προσθέτει αντίστοιχο ποσοστό του μέγιστου variance στο υπολογιζόμενο εμπειρικό variance κάθε feature, για να αποφευχθούν τέτοια φαινόμενα αστάθειας. Να ακολουθήσουμε κι εμείς παρόμοια τεχνική;

Ευχαριστώ

**Comment by georgepar** [comment]
Αν απειρίζεται ο παρονομαστής, μπορείτε να προσθέσετε μια μικρή σταθερά.

Δεν χρειάζεται πιο πολύπλοκη μοντελοποίηση.

**Comment by TsiarasKon** [comment]
Παρατηρήσαμε (με τιμές της τάξεως 1e-10) ότι όσο μεγαλύτερη κάνουμε αυτή τη σταθερά, τόσο καλύτερο προκύπτει το accuracy του ταξινομητή μας. Κάτι λογικό, με βάση και το αποτέλεσμα του Βήματος 16.
Όσον αφορά όμως τα Βήματα 15β και 15γ, όπου χρειαζόμαστε κάποιο συγκεκριμένο σκορ το οποίο και να αντιπαραθέσουμε με το αντίστοιχο του NB της sklearn, τι τιμή περιμένετε να διαλέξουμε για τη σταθερά αυτή; 
Θεωρούμε πώς δεν θα ήταν επιτρεπτό να την αντιμετωπίσουμε σαν hyperparameter και δοκιμάζοντας πλήθος τιμών να κρατήσουμε αυτή που μας παράγει το βέλτιστο accuracy καθώς αυτό προκύπτει εισάγοντας bias από το test set.

**Comment by georgepar** [comment]
Έχετε μεγάλη απόκλιση για διαφορετικές τιμές της σταθεράς; Αν ναι είναι ενδιαφέρον να το σχολιάσετε στην αναφορά.
Σωστά κάνετε και δε τη θεωρείτε σαν υπερπαράμετρο. 
Προτείνω να τη φιξάρετε στο 1e-5 (που είναι η συνηθισμένη τιμή).
